lib/jsdom/living/nodes/HTMLImageElement-impl.js

```diff
@@ -29,10 +64,10 @@ class HTMLImageElementImpl extends HTMLElementImpl {
   }
 
   get height() {
-    if (this.hasAttribute("height")) {
-      return conversions["unsigned long"](this.getAttribute("height"));
-    }
-    return 0;
+    // Just like on browsers, if no width / height is defined, we fall back on the
+    // dimensions of the internal image data.
+    return this.hasAttribute("height") ?
```

Comment:
BTW, if I create a canvas with a width of 2147483648 on Chrome and then inspect it, it looks like it's not doing any such clamping: `<canvas width="-2147483648" height="1"></canvas>` The with just wraps around as a signed 32 bit integer value.